packages/strapi-utils/lib/hooks.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
```

Comment:
I find it hard to read for very simple logic

Reply:
What part do you find hard to read? (I could split the logic in dedicated files if needed)